src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -2549,19 +2500,11 @@ void Thread::RareEnablePreemptiveGC()
     if (IsAtProcessExit())
         return;
 
-    // EnablePreemptiveGC already set us to preemptive mode before triggering the Rare path.
-    // Force other threads to see this update, since the Rare path implies that someone else
-    // is observing us (e.g. SuspendRuntime).
-
     _ASSERTE (!m_fPreemptiveGCDisabled);
 
     // holding a spin lock in coop mode and transit to preemp mode will cause deadlock on GC
     _ASSERTE ((m_StateNC & Thread::TSNC_OwnsSpinLock) == 0);
 
-    _ASSERTE(!MethodDescBackpatchInfoTracker::IsLockOwnedByCurrentThread() || IsInForbidSuspendForDebuggerRegion());
```

Comment:
Why did you get rid of this assert?

Reply:
It looks like the assert was added fairly recently, after my changes and was somehow removed when I rebased onto master. I will add it back.

---
In reply to: [508042246](https://github.com/dotnet/runtime/pull/34666#discussion_r508042246) [](ancestors = 508042246)

Reply:
Thanks for catching this.

---
In reply to: [508120593](https://github.com/dotnet/runtime/pull/34666#discussion_r508120593) [](ancestors = 508120593,508042246)